dhis2/src/main/java/org/motechproject/dhis2/rest/service/impl/DhisWebServiceImpl.java

```diff
@@ -511,6 +548,57 @@ private DhisDataValueStatusResponse importDataValues(Settings settings, String u
             String msg = String.format("Error parsing response from uri: %s, exception: %s", uri, e.toString());
             statusMessageService.warn(msg, MODULE_NAME);
             throw new DhisWebException(msg, e);
+        } finally {
+            closeResponse(response);
+        }
+    }
+
+    private void closeResponse(CloseableHttpResponse response) {
+        try {
+            if (response != null) {
+                EntityUtils.consume(response.getEntity());
+                response.close();
+            }
+        } catch (IOException e) {
+            LOGGER.error("Error trying to close response: " + e.getMessage());
+        }
+    }
+
+    @PreDestroy
+    public void cleanUp() throws IOException {
+        client.close();
+        poolingHttpClientConnectionManager.close();
+    }
+
+    private final class IdleConnectionMonitorThread extends Thread {
+        private final HttpClientConnectionManager connMgr;
+        private volatile boolean shutdown;
+
+        private IdleConnectionMonitorThread(PoolingHttpClientConnectionManager connMgr) {
```

Comment:
I was following instructions from apache httpclient tutorial that say:
 "The stale connection check is not 100% reliable. The only feasible solution that does not involve a one thread per socket model for idle connections is a dedicated monitor thread used to evict connections that are considered expired due to a long period of inactivity"
https://hc.apache.org/httpcomponents-client-ga/tutorial/html/connmgmt.html 
2.5. Connection eviction policy